src/navigation/search.py

```diff
@@ -93,14 +94,24 @@ class SearchState(State):
 
     def on_enter(self, context) -> None:
         search_center = context.course.current_waypoint()
+
         if not self.is_recovering:
-            self.traj = SearchTrajectory.spiral_traj(
-                context.rover.get_pose().position[0:2],
-                self.SPIRAL_COVERAGE_RADIUS,
-                self.DISTANCE_BETWEEN_SPIRALS,
-                self.SEGMENTS_PER_ROTATION,
-                search_center.tag_id,
-            )
+            if search_center.type.val == WaypointType.POST:
+                self.traj = SearchTrajectory.spiral_traj(
+                    context.rover.get_pose().position[0:2],
+                    self.SPIRAL_COVERAGE_RADIUS,
+                    self.DISTANCE_BETWEEN_SPIRALS,
+                    self.SEGMENTS_PER_ROTATION,
+                    search_center.tag_id,
+                )
+            else:  # water bottle or mallet
+                self.traj = SearchTrajectory.spiral_traj(
+                    context.rover.get_pose().position[0:2],
+                    self.SPIRAL_COVERAGE_RADIUS / 2,
+                    self.DISTANCE_BETWEEN_SPIRALS / 2,
```

Comment:
well actually our visibility might be different since this is object detection not tag detection. we should probably just add separate constants for object search radius (which is 10m according to the urc rules) and distance between spirals which we can tune in testing